### PR TITLE
Go: Update query help for `go/path-injection` to include example fixes.

### PR DESCRIPTION
**Pull request by max-schaefer**
Closely mimics the changes for Java in https://github.com/github/codeql/pull/15409.

### CODE REVIEW

File: go/ql/src/Security/CWE-022/TaintedPathGood.go
Comment (mbg) [inline go/ql/src/Security/CWE-022/TaintedPathGood.go:14]: Like in the text, it may be good to highlight here that this is only suitable if `path` is expected to refer to a single filename (in case someone looks at this file independently of the `qhelp` context).

Perhaps it would also be good to use the `PathSeparator` constant, instead of hard-coding `/` and `\\`?

**Review by mbg — COMMENTED** [review]
Seems reasonable. I have one or two small suggestions for the first, new example.

File: go/ql/src/Security/CWE-022/TaintedPathGood.go
Comment (mbg) [reply go/ql/src/Security/CWE-022/TaintedPathGood.go:15]: Perhaps, as a compromise, the check could combine both. Something like:

```go
strings.ContainsAny(path, "/\\") || strings.Contains(path, string(os.PathSeparator)) || strings.Contains(path, "..")
```

or

```go
(dir, filename) := filepath.Split(path); dir != "" || strings.Contains(path, "/") || strings.Contains(path, "..")
```

Comment (max-schaefer) [reply go/ql/src/Security/CWE-022/TaintedPathGood.go:15]: Ah, interesting! Does the query recognise these patterns as sanitisers?

Comment (mbg) [reply go/ql/src/Security/CWE-022/TaintedPathGood.go:15]: It probably _should_, but it looks like neither `ContainsAny` nor performing a check with `Split` is accepted as a sanitiser. For `Contains`, we only seem to accept calls with one of `..`, `../`, or `..\\` as argument.

Comment (max-schaefer) [reply go/ql/src/Security/CWE-022/TaintedPathGood.go:15]: OK, shall we leave the QHelp as it is, then, and tackle the query improvements (and accompanying QHelp rewrites) separately?

Comment (mbg) [reply go/ql/src/Security/CWE-022/TaintedPathGood.go:15]: Sounds OK -- I have created an internal issue to track this.

**Review by mbg — APPROVED** [review]

File: go/ql/src/Security/CWE-022/TaintedPath.qhelp
Comment (subatoi) [inline go/ql/src/Security/CWE-022/TaintedPath.qhelp:67]:
```suggestion
If the input can be a path with multiple components, you can make it safe by verifying
```

Minor: the rest of the doc is second-person

File: go/ql/src/Security/CWE-022/TaintedPath.qhelp
Comment (subatoi) [inline go/ql/src/Security/CWE-022/TaintedPath.qhelp:69]:
```suggestion
You can do this by resolving the input with respect to that directory, and then checking
```

Minor: same comment as above

**Review by subatoi — APPROVED** [review]
Looks good, thank you!

Very minor but I noticed a couple of very minor mistakes in the formatting on unchanged lines—I don't think it's worth changing unless you have time, as eventually all this will be migrated and it won't affect reader comprehension.

> In the second example, it appears that the user is restricted to opening a file within the "user" home directory. However, a malicious user could enter a file name containing special characters. For example, the string "../../etc/passwd" will result in the code reading the file located at "/home/user/../../etc/passwd", which is the system's password file. This file would then be sent back to the user, giving them access to password information.

Would recommend changing a couple of things enclosed in "" to be enclosed in backticks (monospaced) per below, with bold formatting used to indicate places:

> In the second example, it appears that the user is restricted to opening a file within the **`user`** home directory. However, a malicious user could enter a file name containing special characters. For example, the string "../../etc/passwd" will result in the code reading the file located at **`/home/user/../../etc/passwd`**, which is the system's password file. This file would then be sent back to the user, giving them access to password information.

**Comment by max-schaefer** [comment]
@subatoi / @mbg, can I get an approval after the latest round of fixups?

I haven't addressed this yet, since it's generally inconsistent:

> Would recommend changing a couple of things enclosed in "" to be enclosed in backticks (monospaced) per below, with bold formatting used to indicate places:

**Comment by owen-mc** [comment]
@max-schaefer Note that more sanitizers for this query were added in https://github.com/github/codeql/pull/11703, soon after this PR was merged, in case you want to update the query help to use one of those.

**Comment by max-schaefer** [comment]
Thanks! From a quick look I think none of these new sanitisers are general enough to suggest in the QHelp.

**Pull request merged**
